### PR TITLE
Crn 1236 introduce working groups on events calendar page

### PR DESCRIPTION
**Pull request by oratiu**

### CODE REVIEW

File: packages/react-components/src/templates/__tests__/EventsCalendar.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { render } from '@testing-library/react';
+
+import EventsCalendar from '../EventsCalendar';
+
+it('Renders calender list', () => {
```
Comment (lctrt) [inline packages/react-components/src/templates/__tests__/EventsCalendar.tsx:5]:
```suggestion
it('Renders calendar list', () => {
```

File: packages/fixtures/src/calendars.ts [hunk 2]
```diff
@@ -1,101 +1,31 @@
...
-  if (itemIndex > calendarResponses.length) {
-    throw new Error(`Exceeds fixture data limit ${calendarResponses.length}`);
-  }
-  return calendarResponses[itemIndex];
-};
+export const createCalendarResponse = (
+  { group = false, workingGroup = false }: FixtureOptions = {},
+  itemIndex = 0,
+): CalendarResponse => ({
+  id: `calendar-${itemIndex}`,
```
Comment (lctrt) [inline packages/fixtures/src/calendars.ts:12]: 🎉

File: packages/react-components/src/templates/__tests__/EventsCalendar.tsx (hunk 1 above)
Comment (oratiu) [reply packages/react-components/src/templates/__tests__/EventsCalendar.tsx:5]: Woops

**Review by lctrt — APPROVED** [review]

File: apps/crn-server/test/data-providers/calendars.data-provider.test.ts [hunk 3]
```diff
@@ -172,7 +172,12 @@ describe('Calendars data provider', () => {
           order: 'data/name/iv asc',
         },
       );
-      expect(result).toEqual({ total: 1, items: [getCalendarDataObject()] });
+      expect(result).toEqual({
+        total: 1,
+        items: [
+          { ...getCalendarDataObject(), workingGroup: true, group: true },
```
Comment (peterstarling) [inline apps/crn-server/test/data-providers/calendars.data-provider.test.ts:178]: why not update the fixture? It's strange you are having to do this here and not in here https://github.com/yldio/asap-hub/blob/2471f1141b14d49b3ebd2b1aeec95cdf59e994d3/apps/crn-server/test/data-providers/calendars.data-provider.test.ts#L99

File: apps/crn-server/test/entities/working-group.test.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    expect(
+      parseGraphQlWorkingGroup(workingGroupWithCalendars).calendars,
+    ).toEqual([
+      {
+        id: 'hub@asap.science',
+        color: '#B1365F',
+        name: 'ASAP Hub',
+      },
+    ]);
+  });
```
Comment (peterstarling) [inline apps/crn-server/test/entities/working-group.test.ts:36]: if you are going to test the parser separately you might as well add another test for the entire input and output

File: packages/model/src/calendar.ts [hunk 5]
```diff
@@ -57,6 +57,8 @@ export interface CalendarDataObject {
   resourceId?: string | null;
   expirationDate?: number | null;
   version: number;
+  group?: boolean;
```
Comment (peterstarling) [inline packages/model/src/calendar.ts:60]: this doesn't look right - first of all an optional boolean makes no sense to me. Second - are we sure we don't just want to return the list of groups and workingGroups associated with it? Weren't you supposed to remove the calendars from the list of working-groups which are inactive?

**Review by peterstarling — CHANGES_REQUESTED** [review]
I think you have missed an important edge case that we spoke about. By leaving the functionality that removes a calendar if it belongs to an inactive group you are effectively removing any calendar that belongs to both an inactive group and a working-group. Similarly you should make sure that for a calendar that is linked with more than one group and one of them is inactive the calendar is still present (something that should have been captured previously but worth having another look).

Although if you think such edge case does not need to be handled and we are ok to just remove the calendar if it belongs to two - you should still write a test for it so we know it was intended.

File: apps/crn-server/test/entities/working-group.test.ts (hunk 4 above)
Comment (oratiu) [reply apps/crn-server/test/entities/working-group.test.ts:47]: Wrote a test for the entire input and output instead as it also contains the calendar part

File: apps/storybook/src/CalendarList.stories.tsx [hunk 6]
```diff
@@ -14,7 +14,7 @@ export const Normal = () => (
   <CalendarList
     page={select<ComponentProps<typeof CalendarList>['page']>(
       'Page',
-      ['event', 'calendar', 'group'],
+      ['event', 'calendar', 'group', 'calendar-working-group'],
```
Comment (kmaid) [inline apps/storybook/src/CalendarList.stories.tsx:17]: Why is this calendar-working-group? Shouldn't it just be working-group?

File: apps/storybook/src/EventsCalendar.stories.tsx [hunk 7]
```diff
@@ -1,16 +1,21 @@
...
-  calendars: calendars.slice(0, number('Calendar item', 1, { min: 0, max: 6 })),
-});
-
-export const Normal = () => <EventsCalendar {...props()} />;
+export const Normal = () => (
+  <EventsCalendar
+    calendars={
+      createListCalendarResponse(number('Calendar item', 1, { min: 0 }), {
+        activeGroups: boolean('Calendar connected to Group', true),
+        incompleteWorkingGroups: boolean(
```
Comment (kmaid) [inline apps/storybook/src/EventsCalendar.stories.tsx:14]: Should this be activeWorkingGroups to be consistent?

File: packages/fixtures/src/calendars.ts [hunk 8]
```diff
@@ -1,101 +1,34 @@
...
-};
+export const createCalendarResponse = (
+  {
+    activeGroups = false,
+    incompleteWorkingGroups = false,
+  }: FixtureOptions = {},
+  itemIndex = 0,
+): CalendarResponse => ({
+  id: `calendar-${itemIndex}`,
+  color: '#B1365F',
```
Comment (kmaid) [inline packages/fixtures/src/calendars.ts:16]: Can we vary the colour?

File: packages/fixtures/src/working-groups.ts [hunk 9]
```diff
@@ -58,10 +60,13 @@ export const createWorkingGroupResponse = (
...
   lastModifiedDate: '2020-11-09T20:36:54Z',
   externalLink: `https://www.example.com/working-group-${itemIndex}`,
-  deliverables: createDeliverables(options?.deliverables ?? 1),
+  deliverables: createDeliverables(deliverables ?? 1),
   complete: false,
-  members: createWorkingGroupMembers(options?.members ?? 1),
-  leaders: createWorkingGroupLeaders(options?.members ?? 1),
+  members: createWorkingGroupMembers(members ?? 1),
+  leaders: createWorkingGroupLeaders(members ?? 1),
+  calendars: Array.from({ length: calendarsCount }, (_, index) =>
```
Comment (kmaid) [inline packages/fixtures/src/working-groups.ts:67]: I don't think you will get your default if you set deliverables to 1 because calendars count will be undefined. If thats not the case can you change teh others to be consistent with this approach

File: apps/storybook/src/EventsCalendar.stories.tsx (hunk 7 above)
Comment (oratiu) [reply apps/storybook/src/EventsCalendar.stories.tsx:14]: Well it depends, the actual property is "complete" which means the working group is no longer active, I tried keeping consistency with how it's seen on Squidex

File: apps/storybook/src/CalendarList.stories.tsx (hunk 6 above)
Comment (oratiu) [reply apps/storybook/src/CalendarList.stories.tsx:17]: That will happen once we add the calendar list on the Working Group page, basically "calendar" is meant for the Calendar page, where we showed the Groups Calendar list. I think I should rather rename "calendar" to "calendar-group"

File: packages/react-components/src/organisms/CalendarList.tsx [hunk 10]
```diff
@@ -54,67 +70,113 @@ const orderList = css({
...
+  const [showMore, setShowMore] = useState(false);
+  const displayShowMoreButton = calendars.length > truncateFrom;
+
+  return (
+    <div css={containerStyles}>
+      <Card padding={false}>
+        <div css={headerStyles}>
+          <Headline3 noMargin>
+            Subscribe to{' '}
+            {page === 'group'
```
Comment (kmaid) [inline packages/react-components/src/organisms/CalendarList.tsx:102]: I think this would be better done with a Record object. There are too many ternary operators to follow.

`
{
'group' : {
	headline: 'aaa',
	description: 'bbb',
	}
}

`

File: packages/react-components/src/organisms/CalendarList.tsx [hunk 11]
```diff
@@ -54,67 +70,113 @@ const orderList = css({
...
+        )}
+        {displayShowMoreButton && (
+          <div css={showMoreStyles}>
+            <Button linkStyle onClick={() => setShowMore(!showMore)}>
+              View {showMore ? 'Less' : 'More'}
+            </Button>
+          </div>
+        )}
+      </Card>
+      {showBottomMessage && (
```
Comment (kmaid) [inline packages/react-components/src/organisms/CalendarList.tsx:164]:
```suggestion
      {showSupportText && (
```
or something

File: packages/react-components/src/templates/EventsCalendar.tsx [hunk 12]
```diff
@@ -13,10 +13,27 @@ type EventsCalendarPageProps = {
...
+    <GoogleCalendar calendars={calendars} />
+    <CalendarList
+      page="calendar"
+      calendars={calendars.filter(
+        (calendar) =>
+          calendar.activeGroups === true ||
+          (calendar.activeGroups === false &&
+            calendar.incompleteWorkingGroups === false),
+      )}
+      showBottomMessage={false}
```
Comment (kmaid) [inline packages/react-components/src/templates/EventsCalendar.tsx:29]:
```suggestion
	hideSupportText
```

File: apps/crn-server/src/data-providers/calendars.data-provider.ts [hunk 13]
```diff
@@ -156,4 +173,12 @@ export const parseGraphQlCalendarToDataObject = (
   version: item.version,
   expirationDate: item.flatData.expirationDate,
   syncToken: item.flatData.syncToken,
+  activeGroups:
+    item.referencingGroupsContents?.findIndex(
+      (group) => group.flatData.active === true,
+    ) !== -1,
+  incompleteWorkingGroups:
+    item.referencingWorkingGroupsContents?.findIndex(
```
Comment (kmaid) [inline apps/crn-server/src/data-providers/calendars.data-provider.ts:181]: You should really use .some

File: apps/storybook/src/CalendarList.stories.tsx (hunk 6 above)
Comment (kmaid) [reply apps/storybook/src/CalendarList.stories.tsx:17]: Starting to wonder if you should just inject the title and description to simplify the whole thing. How many times is the component used? 4 times?

File: packages/react-components/src/organisms/__tests__/CalendarList.test.tsx [hunk 14]
```diff
@@ -1,12 +1,14 @@
...
-import { render } from '@testing-library/react';
+import { fireEvent, render } from '@testing-library/react';
+import { createListCalendarResponse } from '@asap-hub/fixtures';
 
 import CalendarList from '../CalendarList';
 
 const props: ComponentProps<typeof CalendarList> = {
   calendars: [],
-  page: 'calendar',
+  title: 'Subscribe to this Interest Groups Calendar',
```
Comment (kmaid) [inline packages/react-components/src/organisms/__tests__/CalendarList.test.tsx:9]:
```suggestion
  title: '',
```
You don't want to include anything here that can be asserted on to encourage people to set the data within the tests for clarity,

**Review by kmaid — APPROVED** [review]

File: packages/model/src/calendar.ts [hunk 15]
```diff
@@ -72,6 +74,8 @@ export interface CalendarResponse {
   id: string;
   name: string;
   color: GoogleLegacyCalendarColor;
+  activeGroups?: boolean;
```
Comment (peterstarling) [inline packages/model/src/calendar.ts:77]:
```suggestion
  hasActiveGroups?: boolean;
```

File: packages/model/src/calendar.ts [hunk 16]
```diff
@@ -72,6 +74,8 @@ export interface CalendarResponse {
...
+  incompleteWorkingGroups?: boolean;
```
Comment (peterstarling) [inline packages/model/src/calendar.ts:78]:
```suggestion
  hasIncompleteWorkingGroups?: boolean;
```

**Review by peterstarling — CHANGES_REQUESTED** [review]

File: packages/model/src/calendar.ts (hunk 16 above)
Comment (kmaid) [reply packages/model/src/calendar.ts:78]: I really don't like prefixing has on variables. I think its very easily inferred from the type and overly verbose. I think it also encourages good practices because if we ever needed to return an actual list of groups it wouldn't allow you to also maintain this flag with the same variable name

Comment (peterstarling) [reply packages/model/src/calendar.ts:78]: i agree we should not prefix properties such as `active` with prefixes like `has` or `is`but i would actually use your example as a reason why we should do it here - if we ever want to return an actual list of `groups` or `activeGroups` we will not be able to as the name will already be taken. 

It is very confusing to have a `groups` property which is not an array of groups (like everywhere else) but a boolean. 

We want to be consistent and to leave the interfaces open for extension, not for modification (see what i did here)

**Review by peterstarling — CHANGES_REQUESTED** [review]
